src/services/shopify-client.ts

```diff
@@ -21,3 +24,45 @@ export type ShopifyErrorResult = { cannotConnectToAPI: true };
 export interface ShopifyClient {
   getShopifyData(): Promise<Result<ConnectedShop[], ShopifyErrorResult>>;
 }
+
+export class HttpShopifyClient implements ShopifyClient {
+  private readonly axios: AxiosInstance;
+  private readonly baseUrl: string;
+  private readonly connectionDataRef: RefObject<AppSession>;
+
+  constructor({
+    axiosStatic,
+    baseUrl,
+    connectionDataRef,
+  }: {
+    axiosStatic: AxiosStatic;
+    baseUrl: string;
+    connectionDataRef: RefObject<AppSession>;
+  }) {
+    this.baseUrl = baseUrl;
+    this.axios = axiosStatic.create({
+      baseURL: this.baseUrl,
+    });
+    this.connectionDataRef = connectionDataRef;
+  }
+
+  private getShopifyConnectionData(): ShopifyConnectionData {
+    const connectionData = this.connectionDataRef.current;
+    if (!connectionData || connectionData.status !== 'authenticated' || !connectionData.jwtToken) {
+      throw new Error('Shopify connection data is not available');
+    }
+    return connectionData;
+  }
+
+  public async getShopifyData(): Promise<Result<ConnectedShop[], ShopifyErrorResult>> {
+    const { jwtToken } = this.getShopifyConnectionData();
```

Comment:
Should the potential exception here be wrapped in a `ShopifyErrorResult`?

Reply:
Perhaps :) This as in Datahub gets the jwtToken and that's it.

Reply:
Oh, _DataHub Client_ is not exception free, I think that you should use _Doppler Client_ as example: https://github.com/FromDoppler/doppler-webapp/blob/57af926f370aeee1c7bb2ad83bce813078ab6179/src/services/doppler-legacy-client.ts#L369-L373